### PR TITLE
WB-835: Update withActionScheduler to match webapp's HOC pattern

### PR DESCRIPTION
**Pull request by kevinbarabash**
Using this pattern adds support for default props and ref forwarding.  This will allow us to compose `withActionScheduler` with the HOCs defined in webapp.  It also means that all of our HOCs from wonder-blocks and webapp will have a consistent API as well as consistent behavior.

### CODE REVIEW

File: packages/wonder-blocks-timing/components/with-action-scheduler.test.js [hunk 1]
```diff
@@ -22,4 +24,21 @@ describe("withActionScheduler", () => {
...
+            }
+        }
+        const TestComponent = withActionScheduler(Component);
+        let ref: mixed = null;
+
+        // Act
+        mount(<TestComponent ref={(node) => (ref = node)} />);
+
+        // Assert
+        expect(ref).toBeInstanceOf(Component);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-timing/components/with-action-scheduler.test.js]: Nice test!

File: packages/wonder-blocks-timing/util/types.js [hunk 2]
```diff
@@ -230,15 +230,22 @@ export interface IScheduleActions {
...
      * intervals, timeouts, and animation frame requests.
      */
     schedule: IScheduleActions,
 |};
+
+/**
+ * Extends the given props with props that the `withActionScheduler` higher
+ * order component will inject.
+ */
+export type WithActionScheduler<TOwnProps: {...}> = {|
```
Comment (somewhatabstract) [inline packages/wonder-blocks-timing/util/types.js]: Is this just to make sure this isn't a breaking change?

Comment (kevinbarabash) [reply packages/wonder-blocks-timing/util/types.js]: Yes, exactly.  🙂

**Comment by kevinbarabash** [comment]
@jeresig this shouldn't be breaking at least I intended it not to be since we're still exporting the same types we were before.  I can try using `yarn link` to verify whether or not it is actually breaking before publishing.

**Comment by kevinbarabash** [comment]
I guess it is technically breaking since it'll fix a couple of `$FlowFixMe`s in webapp.  I think I should still give it only a minor bump because it's more like a bug fix, it's not really changing the API of wonder-blocks-timing.

**Comment by jeresig** [comment]
@kevinbarabash ok, that's fine - thank you for checking! A minor version bump seems ok to me.

**Comment by kevinbarabash** [comment]
Sorry if I made it sound like I had already checked with `yarn link` last night.  I knew that there were going to be issues with `$FlowFixMe`s.

I have now checked with `yarn link` and there are some additional issues:
- `withActionScheduler` now takes two generic params, but flow can infer both so I'll have to update all calls to `withActionScheduler` to remove the type params.
- flow gets confused with `yarn link` so I had to restart flow after linking.  Even then it still couldn't resolve "../.../../utils/testing/mount.js" even though it was on disk at the absolute path it said it couldn't find it at.  I think maybe the issue is that that path is above the folder that was symlinked by yarn link.

I still think a minor version bump is appropriate.  I'll be publishing today and will update wonder-blocks in webapp as part of the work I'm doing off the `no-ka` branch.

**Pull request merged**